# HG changeset patch
# User Rik <rik@octave.org>
# Date 1344465232 25200
#      Wed Aug 08 15:33:52 2012 -0700
# Node ID 3e5b00d964ad301876987a099ad656212da5347d
# Parent  9087c2816136109f0a87f0622d10c40e892338a4
Use $(top_builddir)/libgnu for parallelism with $(top_srcdir)/libgnu

* libcruft/Makefile.am, liboctave/Makefile.am, src/Makefile.am:
Use $(top_builddir)/libgnu for parallelism with $(top_srcdir)/libgnu

diff --git a/libcruft/Makefile.am b/libcruft/Makefile.am
--- a/libcruft/Makefile.am
+++ b/libcruft/Makefile.am
@@ -15,21 +15,21 @@
 # for more details.
 #
 # You should have received a copy of the GNU General Public License
 # along with Octave; see the file COPYING.  If not, see
 # <http://www.gnu.org/licenses/>.
 
 include $(top_srcdir)/build-aux/common.mk
 
+AUTOMAKE_OPTIONS = subdir-objects
+
 ## Search local directories before those specified by the user.
 AM_CPPFLAGS = \
-  -I../libgnu -I$(top_srcdir)/libgnu
-
-AUTOMAKE_OPTIONS = subdir-objects
+  -I$(top_builddir)/libgnu -I$(top_srcdir)/libgnu
 
 octlib_LTLIBRARIES = libcruft.la
 
 libcruft_la_SOURCES =
 
 nodist_libcruft_la_SOURCES =
 
 libcruft_la_FFLAGS = $(F77_INTEGER_8_FLAG)
@@ -37,17 +37,17 @@ libcruft_la_FFLAGS = $(F77_INTEGER_8_FLA
 libcruft_la_CPPFLAGS = \
   @CRUFT_DLL_DEFS@ \
   $(AM_CPPFLAGS)
 
 include link-deps.mk
 
 libcruft_la_LIBADD = \
   libranlib.la \
-  ../libgnu/libgnu.la \
+  $(top_builddir)/libgnu/libgnu.la \
   $(LIBCRUFT_LINK_DEPS)
 
 # Increment these as needed and according to the rules in the libtool
 # manual:
 libcruft_current = 1
 libcruft_revision = 0
 libcruft_age = 0
 
diff --git a/liboctave/Makefile.am b/liboctave/Makefile.am
--- a/liboctave/Makefile.am
+++ b/liboctave/Makefile.am
@@ -17,17 +17,17 @@
 # You should have received a copy of the GNU General Public License
 # along with Octave; see the file COPYING.  If not, see
 # <http://www.gnu.org/licenses/>.
 
 include $(top_srcdir)/build-aux/common.mk
 
 ## Search local directories before those specified by the user.
 AM_CPPFLAGS = \
-  -I../libgnu -I$(top_srcdir)/libgnu \
+  -I$(top_builddir)/libgnu -I$(top_srcdir)/libgnu \
   -I$(top_srcdir)/libcruft/misc
 
 EXTRA_DIST = \
   config-ops.sh \
   mk-ops.awk \
   mx-op-inc.mk \
   mx-op-src.mk \
   mx-ops \
diff --git a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -15,45 +15,48 @@
 # for more details.
 #
 # You should have received a copy of the GNU General Public License
 # along with Octave; see the file COPYING.  If not, see
 # <http://www.gnu.org/licenses/>.
 
 include $(top_srcdir)/build-aux/common.mk
 
+AUTOMAKE_OPTIONS = subdir-objects
+
 ## Search local directories before those specified by the user.
 AM_CPPFLAGS = \
   -I$(top_srcdir)/libcruft/misc \
   -I../liboctave -I$(top_srcdir)/liboctave \
   -Ioctave-value -I$(srcdir)/octave-value \
   -Iparse-tree -I$(srcdir)/parse-tree \
   -Ioperators -I$(srcdir)/operators \
   -Iinterp-core -I$(srcdir)/interp-core \
   -Iinterpfcn -I$(srcdir)/interpfcn \
   -Icorefcn \
-  -I../libgnu -I$(top_srcdir)/libgnu
-
-AUTOMAKE_OPTIONS = subdir-objects
+  -I$(top_builddir)/libgnu -I$(top_srcdir)/libgnu
 
 octlib_LTLIBRARIES = liboctinterp.la
 
 if AMCOND_BUILD_COMPILED_AUX_PROGRAMS
 bin_PROGRAMS = \
   mkoctfile \
   octave \
   octave-config
 
 mkoctfile_SOURCES =
 nodist_mkoctfile_SOURCES = mkoctfile.cc
-mkoctfile_LDADD = ../libgnu/libgnu.la $(LIBS)
+mkoctfile_LDADD = $(top_builddir)/libgnu/libgnu.la $(LIBS)
 
 octave_config_SOURCES =
 nodist_octave_config_SOURCES = octave-config.cc
-octave_config_LDADD = corefcn/libcorefcn.la ../libgnu/libgnu.la $(LIBS)
+octave_config_LDADD = \
+  corefcn/libcorefcn.la \
+	$(top_builddir)/libgnu/libgnu.la \
+	$(LIBS)
 
 BUILT_SOURCES_EXTRA = \
   mkoctfile.cc \
   octave-config.cc
 else
 bin_PROGRAMS = \
   octave
 
